plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/DefaultThriftMetastoreClientFactory.java

```diff
@@ -97,14 +117,55 @@ public DefaultThriftMetastoreClientFactory(
                 Optional.ofNullable(config.getSocksProxy()),
                 config.getMetastoreTimeout(),
                 metastoreAuthentication,
-                nodeManager.getCurrentNode().getHost());
+                nodeManager.getCurrentNode().getHost(),
+                buildThriftHttpContext(config));
     }
 
     @Override
-    public ThriftMetastoreClient create(HostAndPort address, Optional<String> delegationToken)
+    public ThriftMetastoreClient create(URI uri, Optional<String> delegationToken)
             throws TTransportException
     {
-        return create(() -> createTransport(address, delegationToken), hostname);
+        return create(() -> getTransportSupplier(uri, delegationToken), hostname);
+    }
+
+    private TTransport getTransportSupplier(URI uri, Optional<String> delegationToken)
+            throws TTransportException
+    {
+        switch (uri.getScheme().toLowerCase(ENGLISH)) {
+            case "thrift" -> {
+                return createTransport(HostAndPort.fromParts(uri.getHost(), uri.getPort()), delegationToken);
+            }
+            case "http", "https" -> {
+                return createHttpTransport(uri, thriftHttpContext.orElseThrow(() -> new IllegalArgumentException("Thrift http context is not set")));
+            }
+            default -> throw new IllegalArgumentException("Invalid metastore uri scheme " + uri.getScheme());
+        }
+    }
+
+    private TTransport createHttpTransport(URI uri, ThriftHttpContext httpThriftContext)
+            throws TTransportException
+    {
+        HttpClientBuilder clientBuilder = createHttpClientBuilder(httpThriftContext);
+        clientBuilder.setDefaultRequestConfig(RequestConfig.custom().setConnectTimeout(timeoutMillis).build());
+        return new THttpClient(uri.toString(), clientBuilder.build());
+    }
+
+    private HttpClientBuilder createHttpClientBuilder(ThriftHttpContext httpThriftContext)
+    {
+        HttpClientBuilder httpClientBuilder = HttpClientBuilder.create();
```

Comment:
Could you add the following dependency and inject `OpenTelemetry` instance? 
```xml
        <dependency>
            <groupId>io.opentelemetry.instrumentation</groupId>
            <artifactId>opentelemetry-apache-httpclient-4.3</artifactId>
        </dependency>
```

e.g. 
```java
        HttpClientBuilder httpClientBuilder = ApacheHttpClientTelemetry.builder(openTelemetry).build().newHttpClientBuilder();
```